app/uk/gov/hmrc/customs/notification/repo/NotificationWorkItemRepo.scala

```diff
@@ -210,12 +210,16 @@ class NotificationWorkItemMongoRepo @Inject()(mongo: MongoComponent,
     }
   }
 
-  override def permanentlyFailedByCsIdExists(csid: ClientSubscriptionId): Future[Boolean] = {
-    val selector = csIdAndStatusSelector(csid, PermanentlyFailed)
+  override def permanentlyFailedAndHttp5xxByCsIdExists(csid: ClientSubscriptionId): Future[Boolean] = {
+    val serverErrorCodeMin = 500
+    val selector = and(
+      gte(NotificationWorkItemFields.mostRecentPushPullHttpStatusFieldName, serverErrorCodeMin),
+      csIdAndStatusSelector(csid, PermanentlyFailed)
+    )
 
     collection.find(selector).first().toFutureOption().map {
       case Some(_) =>
-        logger.info(s"Found existing permanently failed notification for client id: $csid")
+        logger.info(s"Found existing permanently failed notification for client id with most recent push/pull HTTP 500: $csid")
```

Comment:
improve log to return what 5XX error rather than just 500 Error

Reply:
Done!